MACM/MACM.py

```diff
@@ -598,38 +598,39 @@ def step(rng_states,inf_idx_Qs,edges_Qs,Qs,inf_idx_Ps,edges_Ps,Ps,p_by_action,me
                 outgoing_messages[influencee_id,outgoing_message_idx,1] = possible_action
                 outgoing_messages[influencee_id,outgoing_message_idx,2] = int(uniq[0] + influencee_id) + (event_number / RECEIVED_INFORMATION_LIMIT)
                 event_number += 1
-                if possible_action != np.int32(Events.creation_idx):
-                    #pick a random message, message structure: influencerID, action, nodeID, parentID, rootID, informationIDs...
-                    #   scan and count valid messages
-                    selected_message_id = -1
-                    valid_msg_count = 0
-                    for msg_id in range(len(messages) - 1):
-                        if messages[influencee_id, msg_id, 5] >= 0 and messages[influencee_id, msg_id, 5] <= NUM_UNIQUE_INFO_IDS and messages[influencee_id, msg_id, 0] >= 0 and messages[influencee_id, msg_id, 1] >= 0 and messages[influencee_id, msg_id, 1] < Events.et:
-                            valid_msg_count += 1
-                            selected_message_id = msg_id
-                    #  select a valid message uniformly random
-                    if valid_msg_count > 0:
-                        selected_message_id =  int(xoroshiro128p_uniform_float64(rng_states, influencee_id) * valid_msg_count)
-                        msg_id = 0
-                        while msg_id < (len(messages) - 1):
+                # construct 3,4,5,.. indexes as for a creation event for now
+                outgoing_messages[influencee_id,outgoing_message_idx,3] = int(outgoing_messages[influencee_id,outgoing_message_idx,2]) 
+                outgoing_messages[influencee_id,outgoing_message_idx,4] = int(outgoing_messages[influencee_id,outgoing_message_idx,2]) 
+                outgoing_messages[influencee_id,outgoing_message_idx,5] = int(xoroshiro128p_uniform_float64(rng_states, influencee_id) * NUM_UNIQUE_INFO_IDS)
+                if messages.shape[1] > 0:
```

Comment:
Is this check necessary? If this is false there will be no messages in the entire system?

Reply:
It checks whether there are no messages for anybody. 
A comment was added for clarity.